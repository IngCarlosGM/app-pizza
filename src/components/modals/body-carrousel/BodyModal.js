import React from 'react';
import PropTypes from 'prop-types';
import { Pizzas } from '../../templates/Pizzas';

const BodyModal = ({ storeToShow }) => {

    return (
        <div className="container__modal-body">
            <div className="flex-column">
                <span className="modal-body__text-description">{ storeToShow.description }</span>
                <span className="modal-body__text-description">{ storeToShow.address }</span>
            </div>
            
            <Pizzas pizzas={ storeToShow.products }/>
        </div>
    )
}

BodyModal.propTypes = {
    storeToShow: PropTypes.object.isRequired,
}

export default BodyModal;
