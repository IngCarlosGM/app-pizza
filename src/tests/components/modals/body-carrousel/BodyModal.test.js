import { mount } from "enzyme";
import BodyModal from "../../../../components/modals/body-carrousel/BodyModal";
import dataApp from '../../../data/dataApp.json';

describe('Testing BodyModal component', () => {

    const wrapper = mount(
        <BodyModal 
            storeToShow= { dataApp.response.stores[0] }
        />
    );
    
    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to have a description', () => {
        expect( wrapper.find('span').at(0).text() ).toBe( dataApp.response.stores[0].description );
    });

    test('should to have a description', () => {
        expect( wrapper.find('span').at(1).text() ).toBe( dataApp.response.stores[0].address );
    });

    test('should to have a description', () => {
        expect( wrapper.exists('Pizzas') ).toBe( true );
    });
    
});
