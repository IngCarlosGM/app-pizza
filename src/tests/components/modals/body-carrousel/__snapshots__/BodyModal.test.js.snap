// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BodyModal component should render correctly 1`] = `
<div
  className="container__modal-body"
>
  <div
    className="flex-column"
  >
    <span
      className="modal-body__text-description"
    >
      Esta pizzeria se especializa en ingredientes salados
    </span>
    <span
      className="modal-body__text-description"
    >
      Calle 1 #2-3
    </span>
  </div>
  <div
    className="grid-pizzas mt-third"
  >
    <div
      className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
    >
      <div
        className="item-store"
      >
        <img
          alt="Logo pizza"
          className="img-item-store"
          src="../assets/pizzas/pizza-1.png"
        />
      </div>
      <div
        className="flex-column"
      >
        <span
          className="title-item-store"
        >
          Pizza de Pollo
        </span>
      </div>
    </div>
    <div
      className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
    >
      <div
        className="item-store"
      >
        <img
          alt="Logo pizza"
          className="img-item-store"
          src="../assets/pizzas/pizza-2.png"
        />
      </div>
      <div
        className="flex-column"
      >
        <span
          className="title-item-store"
        >
          Piza Pollo y Champiñones
        </span>
      </div>
    </div>
    <div
      className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
    >
      <div
        className="item-store"
      >
        <img
          alt="Logo pizza"
          className="img-item-store"
          src="../assets/pizzas/pizza-3.png"
        />
      </div>
      <div
        className="flex-column"
      >
        <span
          className="title-item-store"
        >
          Pizza Vegetales
        </span>
      </div>
    </div>
    <div
      className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
    >
      <div
        className="item-store"
      >
        <img
          alt="Logo pizza"
          className="img-item-store"
          src="../assets/pizzas/pizza-4.png"
        />
      </div>
      <div
        className="flex-column"
      >
        <span
          className="title-item-store"
        >
          Pizza Vegetales y Atun 
        </span>
      </div>
    </div>
    <div
      className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
    >
      <div
        className="item-store"
      >
        <img
          alt="Logo pizza"
          className="img-item-store"
          src="../assets/pizzas/pizza-5.png"
        />
      </div>
      <div
        className="flex-column"
      >
        <span
          className="title-item-store"
        >
          Pizza 3 Quesos con Champiñones
        </span>
      </div>
    </div>
    <div
      className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
    >
      <div
        className="item-store"
      >
        <img
          alt="Logo pizza"
          className="img-item-store"
          src="../assets/pizzas/pizza-6.png"
        />
      </div>
      <div
        className="flex-column"
      >
        <span
          className="title-item-store"
        >
          Pizza Jamon y Queso
        </span>
      </div>
    </div>
  </div>
</div>
`;
